### PR TITLE
Using archive instead of wget

### PR DESCRIPTION
**Pull request by mikerenfro**
#### Pull Request (PR) description

Working on migrating the `wget` exec types to `archive`. Threw no errors for initial testing on a minimally-installed CentOS 7 system with no `wget` binary available. Tested with downloading and making menu entries for:

- memtest86+
- CentOS 6 (i386, x86_64), 7 (x86_64)
- CoreOS 2303.4.0, 2345.3.0 (amd64)
- Debian 8, 9, 10 (i386, amd64)
- Ubuntu 14.04, 16.04, 18.04 (i386, amd64)

#### This Pull Request (PR) fixes the following issues

N/A

### CODE REVIEW

**Comment by mikerenfro** [comment]
Squashed commits and updated original comment with current scope of testing.

**Comment by bastelfreak** [comment]
@mikerenfro thanks for the awesome work! Can you add tests to check that the resources are in the catalog?

**Comment by mikerenfro** [comment]
I didn't modify anything in the current test framework for resources, so what should I add?

File: spec/acceptance/nodesets/docker/centos-5.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ HOSTS:
     docker_preserve_image: true
     docker_cmd: '["/sbin/init"]'
     docker_image_commands:
-      - 'yum install -y crontabs initscripts iproute openssl sysvinit-tools tar wget which'
+      - 'yum install -y crontabs initscripts iproute openssl sysvinit-tools tar which'
```
Comment (bastelfreak) [inline spec/acceptance/nodesets/docker/centos-5.yml:13]: all those legacy yaml files in the nodesets directory aren't used. They will be dropped in https://github.com/voxpupuli/puppet-pxe/pull/62. please don't update them

File: manifests/tools/memtest.pp [hunk 2]
```diff
@@ -8,11 +8,18 @@
 
   $tftp_root = $pxe::tftp_root
 
-  exec { 'retrieve memtest image':
-    path    => ['/usr/bin', '/usr/local/bin', '/bin'],
-    command => "wget -q -O - ${url} | gunzip > ${tftp_root}/tools/memtest86+",
-    creates => "${tftp_root}/tools/memtest86+",
-    require => Class['Pxe::Tools'],
+  archive { "${tftp_root}/tools/memtest86+-4.20.bin.gz":
```
Comment (bastelfreak) [inline manifests/tools/memtest.pp:11]: We've got a minimal example in https://github.com/voxpupuli/puppet-pxe/blob/master/spec/classes/init_spec.rb for the init class. Can you add the same for the pxe::tools::memtest class to ensure it still compiles? Same for the pxe::syslinux::direct class.

**Comment by bastelfreak** [comment]
the history looks a bit broken here, can you please rebase against our latest master branch? (and don't do a merge commit)

**Comment by mikerenfro** [comment]
> 
> 
> the history looks a bit broken here, can you please rebase against our latest master branch? (and don't do a merge commit)

Yeah, I rebased, but probably did the merge commit. Worst case, I'll wipe this branch and redo the useful changes.

**Comment by bastelfreak** [comment]
thanks for all the work! I fixed the failing test and added coverage for the other resources. acceptance tests would be really helpful here, but kind of a long shot. I think that should be tackled in another PR.

**Pull request merged**